site/content/features/records.md

```diff
@@ -0,0 +1,366 @@
+
+{{< sandbox version="java14" preview="true" mainclass="RecordDemo" >}}
+{{< sandboxsource "RecordDemo.java" >}}
+record PolarPoint(double r, double theta) {
+   public PolarPoint {
+      theta = Math.IEEEremainder(theta, 2 * Math.PI);
+      if (theta < 0) theta += 2 * Math.PI;
+   }
+}
+
+public class RecordDemo {
+   public static void main(String[] args) {
+      PolarPoint p = new PolarPoint(1, 20.5 * Math.PI);
+      System.out.println("Normalized angle: " + p);
+   }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+
+
+You can implement any interfaces:
+
+{{< sandbox version="java14" preview="true" mainclass="RecordDemo" >}}
+{{< sandboxsource "RecordDemo.java" >}}
+record Point(double x, double y) implements Comparable<Point> {
+   public int compareTo(Point other) {
+      int dx = Double.compare(x, other.x);
+      return dx != 0 ? dx : Double.compare(y, other.y);
+   }
+}
+
+public class RecordDemo {
+   public static void main(String[] args) throws InterruptedException {
+      Point[] points = {
+         new Point(4, 3),
+         new Point(3, 5),
+         new Point(3, 4)
+      };
+      java.util.Arrays.sort(points);
+      System.out.println(java.util.Arrays.toString(points));
+   }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+```
+```
+
+Parameterized records—no problem:
+
+{{< sandbox version="java14" preview="true" mainclass="RecordDemo" >}}
+{{< sandboxsource "RecordDemo.java" >}}
+record Point<T>(T x, T y) {}
+
+public class RecordDemo {
+   public static void main(String[] args) throws InterruptedException {
+      var p = new Point<Double>(3.0, 4.0);
+      System.out.println("Double coordinates: " + p);
+      System.out.println("String coordinates: " + new Point<String>("three", "four"));
+      System.out.println("Point coordinates: " + new Point<Point<Double>>(p, p));
+   }
+}
+{{< /sandboxsource >}}
+{{< /sandbox >}}
+
+What You Can't Do
+-----------------
+
+Most importantly, records cannot have any instance variables other than the “record components”—the variables declared with the canonical constructor. The state of a record object is entirely determined by the record components.
+
+You cannot extend a record—it is implicitly `final`.
+
+A record cannot extend another class, not even another record. (Any record implicitly extends `java.lang.Record`, just like any enumerated type implicitly extends `java.lang.Enum`. The `Record` superclass has no state and only abstract `equals`, `hashCode`, and `toString` methods.)
+
+There are no “inner records”. A record that is defined inside another class is automatically `static`. That is, it doesn't have a reference to its enclosing class (which would be an additional instance variable).
+
+The canonical constructor cannot throw checked exceptions:
+
+```
+record SleepyPoint(double x, double y) {
+   public SleepyPoint throws InterruptedException { // Error
+     Thread.sleep(1000); 
+   }
+}
+```
+
+Reflection
+==========
```

Comment:
This should be probably h2 not h1? So 

    Reflection
    ----------

Reply:
Yes